app/code/Magento/InventoryShipping/Plugin/Sales/ResourceModel/Order/Shipment/LoadSourceForShipmentPlugin.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryShipping\Plugin\Sales\ResourceModel\Order\Shipment;
+
+use Magento\Framework\Model\AbstractModel;
+use Magento\InventoryShipping\Model\ResourceModel\ShipmentSource\GetSourceCodeByShipmentId;
+use Magento\Sales\Model\ResourceModel\Order\Shipment as ShipmentResource;
+use Magento\Sales\Api\Data\ShipmentExtensionFactory;
+
+class LoadSourceForShipmentPlugin
+{
+    /**
+     * @var ShipmentExtensionFactory
+     */
+    private $shipmentExtensionFactory;
+
+    /**
+     * @var GetSourceCodeByShipmentId
+     */
+    private $getSourceCodeByShipmentId;
+
+    /**
+     * @param ShipmentExtensionFactory $shipmentExtensionFactory
+     * @param GetSourceCodeByShipmentId $getSourceCodeByShipmentId
+     */
+    public function __construct(
+        ShipmentExtensionFactory $shipmentExtensionFactory,
+        GetSourceCodeByShipmentId $getSourceCodeByShipmentId
+    ) {
+        $this->shipmentExtensionFactory = $shipmentExtensionFactory;
+        $this->getSourceCodeByShipmentId = $getSourceCodeByShipmentId;
+    }
+
+    /**
+     * @param ShipmentResource $subject
+     * @param ShipmentResource $result
+     * @param AbstractModel $shipment
+     * @return ShipmentResource
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterLoad(
+        ShipmentResource $subject,
+        ShipmentResource $result,
+        AbstractModel $shipment
+    ) {
+        $shipmentExtension = $shipment->getExtensionAttributes();
+        if (empty($shipmentExtension)) {
+            $shipmentExtension = $this->shipmentExtensionFactory->create();
+        }
+        $sourceCode = $this->getSourceCodeByShipmentId->execute((int)$shipment->getId());
+        if (!empty($sourceCode)) {
```

Comment:
btw we need to write Data upgrade script and assign to all the existing shipments (already created)
Source Code = 'default' (default Source).

in this case you no need to have this condition `if (!empty($sourceCode)) {`